lib/puppet/provider/package/pip.rb

```diff
@@ -56,11 +56,8 @@ def query
   # cache of PyPI's package list so this operation will always have to
   # ask the web service.
   def latest
-    client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
-    client.http_header_extra = {"Content-Type" => "text/xml"}
-    client.timeout = 10
-    result = client.call("package_releases", @resource[:name])
-    result.first
+    result = JSON.parse(open("https://pypi.python.org/pypi/#{@resource[:name]}/json").read)
```

Comment:
Which... is apparently marked API private. As is `Puppet::Network::HTTP::Pool`.

I suppose since this is a core provider either method is acceptable, but it feels like bad form to be using private APIs in our core providers without exposing similar functionality to 3rd-party ones :/